10x/hca-dcp/wrapper_10x_count.wdl

```diff
@@ -0,0 +1,340 @@
+    mv $primers_file primers.tsv
+
+    python <<CODE
+    inputs = []
+    print('other inputs')
+    with open('${write_objects(other_inputs)}') as f:
+        keys = f.readline().strip().split('\t')
+        for line in f:
+            values = line.strip().split('\t')
+            input = {}
+            for i, key in enumerate(keys):
+                input[key] = values[i]
+            print(input)
+            inputs.append(input)
+
+    print('r1')
+    r1 = ['${sep="', '" r1}']
+    for r in r1:
+        input = {
+            'name': r.split('\t')[-1],
+            'value': r
+        }
+        inputs.append(input)
+
+    print('r2')
+    r2 = ['${sep="', '" r2}']
+    for r in r2:
+        input = {
+            'name': r.split('\t')[-1],
+            'value': r
+        }
+        inputs.append(input)
+
+    print('i1')
+    i1 = ['${sep="', '" i1}']
+    for i in i1:
+        input = {
+            'name': i.split('\t')[-1],
+            'value': i
+        }
+        inputs.append(input)
+
+    print('write inputs.tsv')
+    with open('inputs.tsv', 'w') as f:
+        f.write('name\tvalue\n')
+        for input in inputs:
+            print(input)
+            f.write('{0}\t{1}\n'.format(input['name'], input['value']))
+    print('finished')
+    CODE
+  >>>
+  runtime {
+    docker: "humancellatlas/secondary-analysis-python"
+  }
+  output {
+    File inputs = "inputs.tsv"
+    File primers_tsv = "primers.tsv"
+  }
+}
+
+
+workflow Wrapper10xCount {
+  String bundle_uuid
+  String bundle_version
+
+  File sample_def
+  Int reads_per_file
+  Float subsample_rate
+  Array[Map[String, String]] primers
+  String align
+  File reference_path
+  Int umi_min_qual_threshold
+
+  # Submission
+  File format_map
+  String dss_url
+  String submit_url
+  String reference_bundle
+  String run_type
+  String schema_version
+  String method
+  Int retry_seconds
+  Int timeout_seconds
+
+  call GetInputs {
+    input:
+      bundle_uuid = bundle_uuid,
+      bundle_version = bundle_version,
+      dss_url = dss_url,
+      retry_seconds = retry_seconds,
+      timeout_seconds = timeout_seconds
+  }
+
+  scatter(i in GetInputs.lanes) {
+    call rename_files as prep {
+      input:
+        r1 = GetInputs.r1[i],
+        r2 = GetInputs.r2[i],
+        i1 = GetInputs.i1[i],
+        r1_name = GetInputs.r1_names[i],
+        r2_name = GetInputs.r2_names[i],
+        i1_name = GetInputs.i1_names[i]
+    }
+  }
+
+  call countwdl.count as analysis {
+    input:
+      sample_def = sample_def,
+      r1 = prep.r1_new,
+      r2 = prep.r2_new,
+      i1 = prep.i1_new,
+      sample_id = GetInputs.inputs.sample_id,
+      reads_per_file = reads_per_file,
+      subsample_rate = subsample_rate,
+      primers = primers,
+      align = align,
+      reference_path = reference_path,
+      umi_min_qual_threshold = umi_min_qual_threshold
+  }
+
+  String sample_id = GetInputs.inputs.sample_id
+  File primers_tsv = inputs_for_submit.primers_tsv
```

Comment:
Where does this value get used?

Reply:
Apparently nowhere :) - thanks for catching that.